### ISSUE TITLE
Copy task - Learning on input of length 1

### ISSUE THREAD

**Issue by tristandeleu** [snipsco/ntm-lasagne#4]
As suggested by @adrienball, I ran an experiment to learn the NTM on only length one inputs to see if it could already learn such a simple behavior (even if it overfits). The NTM successfully recovered the length one inputs:
![copy-1](https://cloud.githubusercontent.com/assets/2018752/9998956/21bae724-6094-11e5-982a-31db67fd3bef.png)

When I tested this trained NTM on longer inputs, it consistently failed at recovering the whole sequences (as expected, due to the lack of variety in the input lengths), but generally succeeded to remember the first vector. However some interesting patterns emerged:
- The NTM was sometimes able to recover the first 2 vectors even though it had never seen any inputs larger than two
  ![copy-10-partial](https://cloud.githubusercontent.com/assets/2018752/9999194/4c64eef6-6095-11e5-9b29-9ad011eab63d.png)
- The NTM sometimes repeated this first vector (with some "noise") multiple times. This is an interesting property that has come up frequently enough to be worth investigating on.
  ![copy-10-repeat](https://cloud.githubusercontent.com/assets/2018752/9999072/aaf44bc0-6094-11e5-9a63-f6b2528ea8f4.png)
##### Parameters of the experiment
- NTM layer with FeedForward controller + 1 read head + 1 write head
- Update rule: Graves' RMSprop with `learning_rate=1e-3` (other parameters left as is from his previous paper)
- Activations: ReLu for `[add, key, beta]`, 1 + ReLu for `gamma`, sigmoid for `[gate, dense_output]`, softmax for `shift` 
- Initialization: Uniform Glorot for any weight matrix + Memory init, Zeros for any bias + Hidden state init, EquiProba for weights init (Read & Write)
##### Learning curve

Gray: Cost function, Red: Moving average of the cost function over 500 iterations
![copy-learning-curve](https://cloud.githubusercontent.com/assets/2018752/10018680/05bef8bc-6139-11e5-8534-bdbaa3ba8f93.png)
